common/core/overlaylayer.cpp

```diff
@@ -365,7 +365,9 @@ void OverlayLayer::InitializeState(HwcLayer* layer,
 void OverlayLayer::InitializeFromHwcLayer(
     HwcLayer* layer, ResourceManager* resource_manager,
     OverlayLayer* previous_layer, uint32_t z_order, uint32_t layer_index,
-    uint32_t max_height, uint32_t rotation, bool handle_constraints) {
+    uint32_t max_height, uint32_t rotation, bool handle_constraints,
+    FrameBufferManager *frame_buffer_manager) {
+  fb_manager_ = frame_buffer_manager;
```

Comment:
hmm, how will we set this in case layer is created by NativeSurface ? I dont think we need to store this as variable in Overlaylayer. DisplayPlaneManager can pass the FBManager to NativeSurface and it can pass it to layer when calling SetBuffer (similar to what we do with resource_manager).

Reply:
Hmm. Let me look into this. I don't believe we can directly pass it to SetBuffer since it is being called locally from different places within the file and hence would either need its own pointer to fb_manager or need those methods to also be passed the frame_buffer_manager from elsewhere.

Reply:
Shouldn't be called locally from place other than Initialize related APIs.

Reply:
https://github.com/intel/IA-Hardware-Composer/blob/master/common/core/overlaylayer.cpp#L241

https://github.com/intel/IA-Hardware-Composer/blob/master/common/core/overlaylayer.cpp#L492

I'm referring to those two who would then need to also be passed frame_buffer_manager as well since NativeSurface isn't the only place that is calling it

Reply:
So essentially what I mean is that if we decided to change SetBuffer to have the extra parameter of framebuffermanager then NativeSurface would have to provide it as well as CloneLayer and InitializeState ane etc.  (if we only want overlayer to have it through passed down parameters and not its own variable).

Reply:
ya, makes sense we should do that change.